Fix Featured Projects styling

yieb opened the pull request:
Update styling of the Featured Projects component and the Featured
Projects card to be mobile first

Decrease margins in mobile view of Featured Projects section

Change styling of Featured Projects card to fix overlapping cards and
improve layout while zooming in and out

Fix mobile view of Featured Projects section

Updates also appear to have fixed overlapping cards in Current Projects
section of Our Work page

yieb committed:
Fix Featured Projects styling

Update styling of the Featured Projects component and the Featured
Projects card to be mobile first

Decrease margins in mobile view of Featured Projects section

Change styling of Featured Projects card to fix overlapping cards and
improve layout while zooming in and out

Fix mobile view of Featured Projects section

Updates also appear to have fixed overlapping cards in Current Projects
section of Our Work page

heylookitsme commented:
![image](https://user-images.githubusercontent.com/45301066/183258840-36cac6c1-2218-410f-9505-729c5ce9b2fc.png)

sorry for confusion, but on figma cards are actually left aligned

heylookitsme commented on frontend/src/components/home_page/FeaturedProjectCard.tsx:
```diff
@@ -21,19 +21,21 @@ const FeaturedProjectCard = ({ link, title, date, summary, image, altText }: Fea
-            <a id={styles.cardTitle} href={link}>
-              <h3>{title}</h3>
-            </a>
+        <div id={styles.cardVariableTextContainer}>
```
Note that having two classes `cardTextContainer` and `cardVariableTextContainer` might be a source of future confusion. it would probably be better to opt for something that is more descriptive like "cardDetails" and have the outer one be "cardDetailsContainer"

yieb replied:
@heylookitsme You're right that sounds like a better name. I can create a separate PR for that change.

heylookitsme commented on frontend/src/styles/home/FeaturedProjectCard.module.css:
```diff
@@ -85,19 +68,72 @@
+    height: 200px;
+  }
+
+  #cardVariableTextContainer {
```
im guessing you consulted katherine about this change?

heylookitsme replied:
i.e that it *is* intended behaviour for "see more" link to clamp to bottom of card?

yieb replied:
@heylookitsme This change was done to fix some styling issues. With this implementation, the text above the show more button (the project title + descriptions) can't overflow past the level of the picture. This was causing some ugly styling issues at certain screen sizes before and also was a reason for overlapping text.

heylookitsme commented on frontend/src/styles/home/FeaturedProjectCard.module.css:
```diff
@@ -1,33 +1,19 @@
-
 #cardContainer {
```
throughout: there are many `text-align: center` styling rules for various elements (via id)-- why didn't you just have one rule to style the justification of the text via the container?

yieb replied:
I actually tried that but I _think_ it was messing up the desktop level styling so i got rid of it

heylookitsme commented on frontend/src/styles/home/FeaturedProjectCard.module.css:
```diff
@@ -1,33 +1,19 @@
-  margin-top: 0px;
-  margin-bottom: 3vw;
-  display: flex;
+  flex-direction: column;
```
i believe these first three rules aren't doing anything because the default display is block?
![image](https://user-images.githubusercontent.com/45301066/183259560-0ead1ae8-ebf0-439c-a6f9-d2fc59129461.png)

yieb replied:
Apologies, there might be some redundant styling still left over that I forgot to clean up

yieb replied:
@heylookitsme What do you mean by those first three rules? The ones that your comment indicates are already deleted?? Could you specify exact line numbers?

heylookitsme commented on frontend/src/styles/home/FeaturedProjectCard.module.css:
```diff
@@ -1,33 +1,19 @@
+  margin-bottom: 7vw;
 }
 
 #cardImage {
```
does this actually do anything

yieb replied:
@heylookitsme This was to create some whitespace between the image and the text when in mobile view as shown in the Figma. Without this, the Title text would be hugging the bottom edge of the pictures.
<img width="250" alt="Screenshot" src="https://user-images.githubusercontent.com/52669318/183266449-374eab79-2be3-49de-a387-45564297e46d.png">

heylookitsme reviewed (commented):
it. works. so that's fine. however, in the future/if doing code cleanup, i would consider: 
-  removing styles that dont apply
- thinking about why you use the selection of vw/% for margin and padding you do
- and using broader styling rules instead of lots of ids to accomplish the tasks you're doing. 

also i would consider styling the images in the cards 
![image](https://user-images.githubusercontent.com/45301066/183260082-9551c1a4-fc11-40ad-a59a-e2c15443aeb1.png)
to be centered like they are on 
![image](https://user-images.githubusercontent.com/45301066/183260090-efb9879a-8dee-43d3-ae6e-3b45f12bd616.png)

desktop

Commit:
Left aligned mobile text and decreased margins on featured projects to
be consistent with homepage

- set text-align: left on cardTitleContainer, cardDateContainer,
  cardSummary, cardLinkContainer
- margins for featured projects are now same as currentprojects on
  ourwork page. "as long as the margins are consistent, it doesnt matter
  if theyre large/small" (sic)

yieb commented:
https://github.com/Hack4Impact-UMD/umd-website-redesign/pull/54#pullrequestreview-1064326940

@heylookitsme 
In regards to this comment:
First, in regards to your comment about the images, I forgot to fix it so the images in mobile view are also centered like in desktop before making the PR, that was my bad. I'll add it as a new task on the Kanban just so we remember that's an issue.

About your other comments:

- removing styles that dont apply: I've just been looking through the CSS module file to remove redundant styles so that's why there might be some inactive styling that I didn't remove. The way you do it is by inspecting elements in the loaded page and checking which styles are actually active, right? I should change to checking for inactive styling the way you do it then.

- thinking about why you use the selection of vw/% for margin and padding you do: To my understanding, vw changes size according to the viewport or browser size while % changes size in proportion to the parent element that the styled element is in, so I _think_ most of the styling is correct in regards to this interpretation? You can let me know if there is some questionable use of vw/% that I should change.

- and using broader styling rules instead of lots of ids to accomplish the tasks you're doing: should i add this as a task to the Kanban?
